app/services/sign_in/assertion_validator.rb

```diff
@@ -29,9 +29,13 @@ def validate_service_account_config
     end
 
     def validate_issuer
-      if issuer != audience
-        raise Errors::ServiceAccountAssertionAttributesError.new message: 'Assertion issuer is not valid'
+      error = false
+      if service_account_id && issuer != service_account_id
```

Comment:
I think this can be simplified quite a bit. The reason is that in order to `validate_service_account_config`, either `service_account_id` must exist, or `issuer == service_account_id`. So that logic is already baked in. I think the only remaining validation would be:

`raise Error unless issuer == service_account_id || issuer == audience`